gossip/src/ping_pong.rs

```diff
@@ -192,14 +197,18 @@ impl PingCache {
         now: Instant,
         node: (Pubkey, SocketAddr),
         mut pingf: F,
+        stats: &mut PingCacheStats,
     ) -> Option<Ping<T>>
     where
         T: Serialize,
         F: FnMut() -> Option<Ping<T>>,
     {
         match self.pings.peek(&node) {
             // Rate limit consecutive pings sent to a remote node.
-            Some(t) if now.saturating_duration_since(*t) < self.rate_limit_delay => None,
+            Some(t) if now.saturating_duration_since(*t) < self.rate_limit_delay => {
+                stats.pings_rate_limited += 1;
+                None
+            }
```

Comment:
If you read this code carefully and understand how the ping/pong token lifetime is extended before ttl expiry:
https://github.com/solana-labs/solana/blob/7469ce7ba/gossip/src/ping_pong.rs#L245-L247

you see that, the fact that we get rate limited here does **not** necessary mean that any pull-requests will be dropped. _Even in perfect scenario that all requests are responded, we will hit rate limiting here_, simply because multiple pull-requests arriving around the same time. Gossip indeed sends multiple pull-requests together.

The actual metric we want to look at is how many requests do indeed get dropped because of the expired pong token; and that is already tracked here:
https://github.com/solana-labs/solana/blob/7469ce7ba/gossip/src/cluster_info.rs#L1909-L1911

(obviously we should exclude requests dropped before the very first ping/pong round trip is established because that is a different thing).